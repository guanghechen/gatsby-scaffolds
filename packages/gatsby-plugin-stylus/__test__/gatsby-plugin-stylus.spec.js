const { onCreateWebpackConfig } = require('../gatsby-node')

describe(`gatsby-plugin-stylus`, () => {
  const actions = {
    setWebpackConfig: jest.fn(),
  }

  // loaders "mocks"
  const loaders = {
    miniCssExtract: () => `miniCssExtract`,
    css: args => `css(${JSON.stringify(args)})`,
    postcss: args => `postcss(${JSON.stringify(args)})`,
    null: () => `null`,
  }

  beforeEach(() => {
    actions.setWebpackConfig.mockReset()
  })

  const stylusPlugin = jest.fn().mockReturnValue(`foo`)

  const tests = {
    stages: [`develop`, `build-javascript`, `develop-html`, `build-html`],
    options: {
      'Empty options': {},
      'No options': undefined,
      'Stylus options #1': {
        stylusRule: {
          use: [stylusPlugin()],
        },
        moduleStylusRule: {
          test: /\.custom\.styl$/,
          use: [stylusPlugin()],
          import: [`file.js`, `file2.js`],
        },
      },
      'Stylus options #2': {
        shouldGenerateDts: true,
        shouldUseSourceMap: true,
      },
      'PostCss plugins': {
        postCssPlugins: [`test1`],
      },
      'css-loader use commonjs': {
        cssLoaderOptions: {
          esModule: false,
          modules: {
            namedExport: false,
          },
        },
      },
    },
  }

  tests.stages.forEach(stage => {
    for (const label in tests.options) {
      const options = tests.options[label]
      it(`Stage: ${stage} / ${label}`, () => {
        onCreateWebpackConfig(
          {
            actions,
            loaders,
            stage,
          },
          options,
        )
        expect(actions.setWebpackConfig).toMatchSnapshot()
      })
    }
  })
})
