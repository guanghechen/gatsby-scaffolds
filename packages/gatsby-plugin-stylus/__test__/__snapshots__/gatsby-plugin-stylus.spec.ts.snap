// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-stylus Stage: build-html / Empty options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "import": Array [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                Object {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Empty options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "import": Array [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                Object {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Empty options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "import": Array [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                Object {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "miniCssExtract",
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Empty options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "import": Array [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                Object {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({\\"esModule\\":true,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":true,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":false})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": Array [
                "css({\\"esModule\\":false,\\"import\\":true,\\"importLoaders\\":1,\\"sourceMap\\":false,\\"modules\\":{\\"namedExport\\":false,\\"exportLocalsConvention\\":\\"camelCaseOnly\\"}})",
                "postcss({})",
                Object {
                  "loader": "stylus-loader",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
